core/services/pipeline/orm.go

```diff
@@ -173,7 +173,7 @@ func (o *orm) ProcessNextUnfinishedRun(ctx context.Context, fn ProcessRunFunc) (
 		// IDs.
 		for i, trr := range trrs {
 			for _, tr := range pRun.PipelineTaskRuns {
-				if trr.TaskRun.DotID == tr.DotID {
+				if trr.Task.DotID() == tr.DotID {
```

Comment:
?

Reply:
to kill off TaskRun as it just contains redundant data with Task